include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -33,6 +37,89 @@ namespace execution
 inline namespace __dpl
 {
 
+namespace __internal
+{
+
+#if _ONEDPL_PREDEFINED_POLICIES
+struct __global_instance_tag {};
+#endif
+
+class alignas(sycl::queue) __queue_holder
+{
+    static_assert(sizeof(sycl::queue) >= sizeof(std::uintptr_t));
+    static_assert(sizeof(sycl::queue) % sizeof(std::uintptr_t) == 0);
+    static_assert(alignof(sycl::queue) >= alignof(std::uintptr_t));
+
+    std::uintptr_t __buf[sizeof(sycl::queue)/sizeof(std::uintptr_t)];
+
+    bool __has_queue() const
+    {
+        bool res = (__buf[0] == 0); // If the first size-of-pointer bytes are zeros, we consider there is no valid queue
```

Comment:
```suggestion
        bool res = (__buf[0] != 0); // If the first size-of-pointer bytes are zeros, we consider there is no valid queue
```
In the changes, this conditional got flipped.